### PR TITLE
Refactor servant generated code in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
This is similar to the proxy generated code refactoring from #1038.

### CODE REVIEW

**Review by externl — APPROVED** [review]
Looks good to me!

**Review by pepone — APPROVED** [review]
Looks good

File: cpp/src/slice2cs/Gen.cpp [hunk 1]
```diff
@@ -2750,96 +2825,82 @@ Slice::Gen::DispatcherVisitor::visitOperation(const OperationPtr& operation)
...
+        _out << eb;
+    }
+    else
+    {
+        auto p = returns.front();
+        _out << "(ZeroC.Ice.OutputStream ostr, " << paramTypeStr(p, true) << " value) =>";
+        _out << sb;
+        writeMarshal(operation, true);
+        _out << eb;
+    }
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply cpp/src/slice2cs/Gen.cpp:2903]: Yes, fixed.

File: cpp/src/slice2cs/Gen.cpp [hunk 2]
```diff
@@ -2880,23 +2875,18 @@ Slice::Gen::DispatcherVisitor::writeOutgoingResponseWriter(const OperationPtr& o
...
-        _out << "(ZeroC.Ice.OutputStream ostr, in " << toTupleType(returns, true) << " value) =>";
-        _out << sb;
-        writeMarshal(operation, true);
-        _out << eb;
-    }
     else
     {
-        auto p = returns.front();
-        _out << "(ZeroC.Ice.OutputStream ostr, " << paramTypeStr(p, true) << " value) =>";
+        _out << "(ZeroC.Ice.OutputStream ostr, " << (returns.size() > 1 ? "in " : "") << toTupleType(returns, true)
```
Comment (pepone) [inline cpp/src/slice2cs/Gen.cpp:2888]: Shouldn't we also use `in` when the return  type is a struct?

Comment (bernardnormier) [reply cpp/src/slice2cs/Gen.cpp:2888]: See the comments above in the code. The single-struct return is handled by the "defaultWriter" block.

**Pull request merged**